src/EditorConfigCompletionProvider.ts

```diff
@@ -0,0 +1,108 @@
+import { CompletionItemProvider, CompletionItem, CompletionItemKind, CancellationToken } from 'vscode'
+import { TextDocument, Range, Position } from 'vscode'
+
+class EditorConfigCompletionProvider implements CompletionItemProvider {
+
+	private readonly properties: Property[] = [
+		new Property('root', ['true', 'false'], 'Special property that should be specified at the top of the file outside of any sections. Set to true to stop .editorconfig files search on current file.'),
+		new Property('charset', ['utf-8', 'utf-8-bom', 'utf-16be', 'utf-16le', 'latin1'], 'Set to latin1, utf-8, utf-8-bom, utf-16be or utf-16le to control the character set. Use of utf-8-bom is discouraged.'),
+		new Property('end_of_line', ['lf', 'cr', 'crlf'], 'Set to lf, cr, or crlf to control how line breaks are represented.'),
+		new Property('indent_style', ['tab', 'space'], 'Set to tab or space to use hard tabs or soft tabs respectively.'),
+		new Property('indent_size', ['1', '2', '3', '4', '5', '6', '7', '8'], ' A whole number defining the number of columns used for each indentation level and the width of soft tabs (when supported). When set to tab, the value of tab_width (if specified) will be used.'),
+		new Property('insert_final_newline', ['true', 'false'], 'Set to true to ensure file ends with a newline when saving and false to ensure it doesn\'t.'),
+		new Property('tab_width', ['1', '2', '3', '4', '5', '6', '7', '8'], 'A whole number defining the number of columns used to represent a tab character. This defaults to the value of indent_size and doesn\'t usually need to be specified.'),
+		new Property('trim_trailing_whitespace', ['true', 'false'], 'Set to true to remove any whitespace characters preceding newline characters and false to ensure it doesn\'t.')
+	];
+
+	// =========================================================================
+	// PUBLIC INTERFACE
+	// =========================================================================
+	public provideCompletionItems(document: TextDocument, position: Position, token: CancellationToken): CompletionItem[] {
+		// get text where code completion was activated
+		let rangeFromLineStart = new Range(new Position(position.line, 0), position);
```

Comment:
We can't ask him to do that until we introduce [this guy](https://palantir.github.io/tslint/rules/prefer-const/) into the project.